src/Microsoft.Azure.WebJobs.Host/Queues/MessageHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Queues
+{
+    /// <summary>
+    /// register function executor in the queue
+    /// </summary>
+    /// TODO generic
+    public abstract class MessageHandler
+    {
+        /// <summary>
+        /// Default implementation returns null
+        /// </summary>
+        public virtual CloudQueue PoisonQueue => null;
```

Comment:
Can you comment the invariants on these new public classes:
- who is expected to implement this class?
- What does Poison Queue mean here. We've abstracted away the other queue mechanics, why is this exposed?

Reply:
Agree - we shouldn't have to expose any CloudQueueDetails.

Reply:
this is added because blobtrigger has its own logic specifying which poison queue to use
https://github.com/Azure/azure-webjobs-sdk/blob/e648136df61a6f43ac7bc139739e24df9f2fbfef/src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/BlobListenerFactory.cs#L178-L188
Expose CloudQueue is probably too much, it can be replaced by some sort of `storage interface` (has a write method) so that if user of sharedQueue does not want to have their failed messaged stored in the default host poison queue, they can bring their own alternative